### ISSUE TITLE
Cannot reproduce results using FastText embeddings

### ISSUE THREAD

**Issue by wouterMostard** [tca19/near-lossless-binarization#9]
First of all thank you for the great paper, really like the simplicity in the model! Currently I am trying to reproduce the semantic similarity scores but somehow this doesn't seem to work. What I did:

Use the Wiki 1M 300 vec embeddings from https://fasttext.cc/docs/en/english-vectors.html

train using the parameters as specified in the paper for FastText (10 epochs, all other values default) 

Run your semantic evaluation scores:

```
create_vocab(): 0.008999s
load_vectors(): 0.926019s
Filename     | Spearman | OOV
==============================
WS353.txt    |    0.559 |   0%
MEN.txt      |    0.622 |   0%
RW.txt       |    0.421 |   3%
SimVerb.txt  |    0.238 |   0%
SimLex.txt   |    0.334 |   0%
evaluate(): 0.008135s
```

Maybe I am missing something but the scores for WS353, RW, SimVerb, SimLex are quite a bit lower. Am I using the correct FastText embeddings? I did not perform any sort of normalization or standardization 

Thank you in advance!

**Comment by tca19** [comment]
Hello @wouterMostard ,

Thank you for your interest in my work. The fasttext embeddings you are using are not the correct ones. I started to develop this model in 2018, so I used the pre-trained fasttext embedding of this era. At that time, the fasttext.cc website did not even existed (links to download pre-trained vectors were still in the README of the fasttext repo). The link to download those vector is dead, but hopefully I still have them on my server, so I have uploaded them again for you. You can download the same pre-trained fasttext vectors as me at this link [1] (767 MB, bzip2 compressed). I think those vectors were trained only on the entire 2018 Wikipedia version.

Then, if you want to replicate my scores, using the default values is not a good idea. I only indicated default values so one can binarize word vectors in a very simple way, but to get better results you will have to tune the hyperparameters. I don't remember the hyperparameters I have used, but I tried a basic example with:
``` ./binarize -input ft-300d.vec -output replicate-ft-256.bin -n-bits 256 -lr-rec 0.001 -lr-reg 0.004 -batch-size 75 -epoch 5```
and here are the results I obtain, which are much closer to the results indicated in the paper:
```
 $ ./similarity_binary replicate-ft-256.bin
create_vocab(): 0.012558s
load_vectors(): 0.560360s
Filename     | Spearman | OOV
==============================
MEN.txt      |    0.732 |   0%
WS353.txt    |    0.668 |   0%
SimLex.txt   |    0.416 |   0%
RW.txt       |    0.506 |   3%
SimVerb.txt  |    0.346 |   0%
evaluate(): 0.005977s
```

In case you absolutely need to have the same scores as in my paper (for accurate comparison with respect to my method), I also have uploaded the binarized fasttext vectors of 256 bits [2] (39 MB, bzip2 compressed). Their results are (which correspond to the scores indicated in the paper):
```$ ./similarity_binary ft-256bits.vec
create_vocab(): 0.015633s
load_vectors(): 0.565324s
Filename     | Spearman | OOV
==============================
MEN.txt      |    0.760 |   0%
WS353.txt    |    0.701 |   0%
SimLex.txt   |    0.447 |   0%
RW.txt       |    0.526 |   3%
SimVerb.txt  |    0.338 |   0%
evaluate(): 0.006044s
```

[1] https://mega.nz/file/UowgDDDY#mlpplNKDppEWlzr8jNHrgt4p_bzDqOQFbYhqJGAyw_c
[2] https://mega.nz/file/M8gXFCQJ#HkCeKeTiAhttPobiW8E5e2O89P2vTHg2D-l00-g9mcE
